Check warnings

### What feature do you want to see added?

There are multiple warnings visible within CI build, see e.g. https://ci.jenkins.io/job/Plugins/job/JiraTestResultReporter-plugin/job/master/51/console

> 16:36:43  [ERROR] Medium: org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor.templates is a mutable collection which should be package protected [org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor] At JiraTestDataPublisher.java:[line 367] MS_MUTABLE_COLLECTION_PKGPROTECT
> 16:36:43  [ERROR] Medium: Possible null pointer dereference of jiraPublisherJSON in org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor.validateFieldConfigs(String) [org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor, org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor, org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor] Dereferenced at JiraTestDataPublisher.java:[line 646]Null value at JiraTestDataPublisher.java:[line 636]Known null at JiraTestDataPublisher.java:[line 638] NP_NULL_ON_SOME_PATH
> 16:36:43  [ERROR] Medium: Redundant nullcheck of uri, which is known to be non-null in org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor.doValidateGlobal(String, String, String) [org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor] Redundant null check at JiraTestDataPublisher.java:[line 544] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
> 16:36:43  [ERROR] Medium: org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor.validateFieldConfigs(String) makes inefficient use of keySet iterator instead of entrySet iterator [org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor] At JiraTestDataPublisher.java:[line 640] WMI_WRONG_MAP_ITERATOR
> 16:36:43  [ERROR] High: Private readResolve method in org.jenkinsci.plugins.JiraTestResultReporter.JobConfigMapping$JobConfigEntry not inherited by subclasses. [org.jenkinsci.plugins.JiraTestResultReporter.JobConfigMapping$JobConfigEntry] At JobConfigMapping.java:[lines 118-119] SE_PRIVATE_READ_RESOLVE_NOT_INHERITED
> 16:36:43  [ERROR] Medium: Exception is caught when Exception is not thrown in org.jenkinsci.plugins.JiraTestResultReporter.TestToIssueMapping.saveMap(Job, HashMap) [org.jenkinsci.plugins.JiraTestResultReporter.TestToIssueMapping] At TestToIssueMapping.java:[line 77] REC_CATCH_EXCEPTION
> 16:36:43  [ERROR] Medium: Possible null pointer dereference in org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi.doJson(StaplerRequest, StaplerResponse) due to return value of called method [org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi, org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi] Dereferenced at TestToIssueMappingApi.java:[line 53]Known null at TestToIssueMappingApi.java:[line 50] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
> 16:36:43  [ERROR] High: Nullcheck of item at line 53 of value previously dereferenced in org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi.doJson(StaplerRequest, StaplerResponse) [org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi, org.jenkinsci.plugins.JiraTestResultReporter.api.TestToIssueMappingApi] At TestToIssueMappingApi.java:[line 53]Redundant null check at TestToIssueMappingApi.java:[line 55] RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE
> 16:36:43  [ERROR] Medium: org.jenkinsci.plugins.JiraTestResultReporter.config.SelectableArrayFields$SelectableArrayFieldsDescriptor.doFillValueItems(String, String, String) might ignore java.lang.Exception [org.jenkinsci.plugins.JiraTestResultReporter.config.SelectableArrayFields$SelectableArrayFieldsDescriptor, org.jenkinsci.plugins.JiraTestResultReporter.config.SelectableArrayFields$SelectableArrayFieldsDescriptor] At SelectableArrayFields.java:[line 164]At SelectableArrayFields.java:[line 164] DE_MIGHT_IGNORE
> 16:36:43  [ERROR] Medium: Exception is caught when Exception is not thrown in org.jenkinsci.plugins.JiraTestResultReporter.config.SelectableArrayFields$SelectableArrayFieldsDescriptor.doFillValueItems(String, String, String) [org.jenkinsci.plugins.JiraTestResultReporter.config.SelectableArrayFields$SelectableArrayFieldsDescriptor] At SelectableArrayFields.java:[line 164] REC_CATCH_EXCEPTION
> 16:36:43  [ERROR] Medium: org.jenkinsci.plugins.JiraTestResultReporter.config.UserFields is Serializable; consider declaring a serialVersionUID [org.jenkinsci.plugins.JiraTestResultReporter.config.UserFields] At UserFields.java:[lines 36-70] SE_NO_SERIALVERSIONID
> 16:36:43  [ERROR] Medium: org.jenkinsci.plugins.JiraTestResultReporter.restclientextensions.FullStatus defines equals but not hashCode [org.jenkinsci.plugins.JiraTestResultReporter.restclientextensions.FullStatus] At FullStatus.java:[lines 48-52] HE_EQUALS_NO_HASHCODE
> 

at least they should be analysed and maybe fixed.

### Upstream changes

_No response_

### Are you interested in contributing this feature?

_No response_

--- a/src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/config/SelectableArrayFields.java
+++ b/src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/config/SelectableArrayFields.java
@@ -166,7 +166,7 @@ public class SelectableArrayFields extends AbstractFields {
                     } else if (o instanceof NamedEntity) {
                         try {
                             Method m = o.getClass().getMethod("getId");
-                            Object id = (Long) m.invoke(o, null);
+                            Object id = (Long) m.invoke(o, (Object[]) null);
                             if (id != null) {
                                 listBox.add(((NamedEntity) o).getName(), id.toString());
                             }

--- a/src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/config/SelectableArrayFields.java
+++ b/src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/config/SelectableArrayFields.java
@@ -166,7 +166,7 @@ public class SelectableArrayFields extends AbstractFields {
                     } else if (o instanceof NamedEntity) {
                         try {
                             Method m = o.getClass().getMethod("getId");
-                            Object id = (Long) m.invoke(o, (Object[]) null);
+                            Object id = (Long) m.invoke(o);
                             if (id != null) {
                                 listBox.add(((NamedEntity) o).getName(), id.toString());
                             }

